src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs

```diff
@@ -345,5 +361,16 @@ private static bool IsPublic(byte[] keyBlob)
 
             return true;
         }
+
+        private static void CheckSHA3HashAlgorithm(HashAlgorithmName hashAlgorithm)
+        {
+            if (hashAlgorithm == HashAlgorithmName.SHA3_256 ||
+                hashAlgorithm == HashAlgorithmName.SHA3_384 ||
+                hashAlgorithm == HashAlgorithmName.SHA3_512)
+            {
+                // Compat: Windows throws CryptographicException for SHA-3 HashAlgorithmName. So we will here, too.
+                throw new CryptographicException(SR.Format(SR.Cryptography_UnknownHashAlgorithm, hashAlgorithm.Name));
```

Comment:
Just clarifying exceptions:

Are all of the asymmetric algorithms throwing CryptographicException when the parameter is a SHA-3 on an unsupported platform (and it's just the lower level "the algorithm is in the type name" types that PNSE?)?

So long as the story can be explained simply, I'm not sure I have strong opinions what it is.  Just observing a CE here and PNSE earlier.